ratchet_core/src/ws.rs

```diff
@@ -384,38 +384,44 @@ async fn close<S>(
 where
     S: WebSocketStream,
 {
+    trace!("Start close fn. Reason {reason:?}, return with {ret:?}");
+
     let server = framed.is_server();
     match *close_state {
         CloseState::NotClosed => {
-            let mut code = match &reason {
-                Some(reason) => u16::from(reason.code).to_be_bytes(),
-                None => [0; 2],
+            let mut code = match (&reason, &ret) {
```

Comment:
There's a fair amout of code here that is identical to that in split and could probably be a function shared between them.